routers/user/setting/profile.go

```diff
@@ -226,3 +226,73 @@ func Repos(ctx *context.Context) {
 
 	ctx.HTML(200, tplSettingsRepositories)
 }
+
+// PinnedRepoPost response pinned repo settings
+func PinnedRepoPost(ctx *context.Context, form auth.RepoPinnedForm) {
+	names := strings.Split(form.RepoFullName, "/")
```

Comment:
Never just use Split where you don't care about all the results.

```suggestion
	names := strings.SplitN(form.RepoFullName, "/", 3)
```

`SplitN(..., 3)`  is enough - because if there is a 3rd group something has gone wrong.